botocore/exceptions.py

```diff
@@ -475,3 +475,15 @@ class MD5UnavailableError(BotoCoreError):
 
 class MetadataRetrievalError(BotoCoreError):
     fmt = "Error retrieving metadata: {error_msg}"
+
+
+class UndefinedModelAttributeError(Exception):
+    pass
+
+
+class MissingServiceIdError(UndefinedModelAttributeError, BotoCoreError):
```

Comment:
I'm a little hesitant to also subclass from BotoCoreError as well.  It can technically change which `except` class you hit, though I realize how unlikely it is that code is catching `UndefinedModelAttributeError`.

I think the safer change is just to subclass `UndefinedModelAttributeError` for now.

Reply:
Well that's why I had a test that specifically catches `UndefinedModelAttributeError`.

Reply:
It's not about a test, it's about code that's using this.  If I have an except clause that catches both `BotoCoreError` and `UndefinedModelAttributeError`, this PR can potentially change which except clause I hit.

If we're going to do that I think we should have a good reason to do this.  Why does it need to also subclass from `BotoCoreError`.   If it's just so we can utilize it's `__init__` method which does the `self.fmt.format(**kwargs)` I don't feel that's a strong enough rationale.

Reply:
I did end up changing it since I didn't have strong feelings one way or another